.github/workflows/build-and-test-msi.yaml

```
@@ -0,0 +1,104 @@
+        run: |
+          aws --version
+      - uses: actions/checkout@8ade135a41bc03ea155e62e844d188df1ea18608 # v4.1.0
+        with:
+          ref: ${{ needs.get-tag-name.outputs.tag }}
+          fetch-depth: 0
+          persist-credentials: false
+          submodules: recursive
+      - name: Set output variables
+        id: vars
+        run: |
+          $has_creds="${{ github.event_name == 'push' || github.repository == github.event.pull_request.head.repo.full_name }}"
+          echo "has_creds=$has_creds" | Out-File -FilePath $Env:GITHUB_ENV -Encoding utf8 -Append
+          exit 0 # if $has_creds is false, powershell will exit with code 1 and this step will fail
+      - name: configure aws credentials
+        uses: aws-actions/configure-aws-credentials@010d0da01d0b5a38af31e9c3470dbfdabdecca3a # v4.0.1
+        with:
+          role-to-assume: ${{ secrets.ROLE }}
+          role-session-name: windows-msi
+          aws-region: ${{ secrets.REGION }}
+      - name: Remove Finch VM
+        run: |
+          wsl --list --verbose
+          wsl --shutdown
+          wsl --unregister lima-finch
+          wsl --list --verbose
+      - name: Clean up previous files
+        run: |
+          Remove-Item C:\Users\Administrator\.finch -Recurse -ErrorAction Ignore
+          Remove-Item C:\Users\Administrator\AppData\Local\.finch -Recurse -ErrorAction Ignore
+          make clean
+          cd deps/finch-core && make clean
+      - name: Build project
+        run: |
+          make FINCH_ROOTFS_LOCATION_ROOT=/__INSTALLFOLDER__
+      - name: generate msi
+        run: |
+          $version="${{ needs.get-tag-name.outputs.tag }}"
+          $version=$version.TrimStart("v")
+
+          if ($version -match '^[0-9]+\.[0-9]+\.[0-9]+$') {
+            Write-Host "Version matches format: $version"
+          }
+          else {
+            Write-Host "Version $version doesn't match format. Using default: 0.0.0"
+            $version="0.0.0"
```

should we fail here instead of falling back to "0.0.0" ? I don't see a case for uploading 0.0.0 versions of the MSI unless there's a testing use case

---

No, it is for testing purpose for not a official release. It is different than mac os because mac os can have hash as version but windows can't